pkg/controller/atlascluster/atlascluster_controller.go

```diff
@@ -108,7 +108,7 @@ func (r *AtlasClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		WithEventFilter(watch.CommonPredicates()).
 		Watches(
 			&source.Kind{Type: &mdbv1.AtlasCluster{}},
-			&watch.DeleteEventHandler{Controller: r},
+			&watch.AtlasResourceEventHandler{Controller: r},
```

Comment:
have changed the name as this handler inherits the generic "create/update" handler as well (and is used for projects below)